server/datastore/mysql/software_installers.go

```diff
@@ -219,6 +234,59 @@ func (ds *Datastore) addSoftwareTitleToMatchingSoftware(ctx context.Context, tit
 	return ctxerr.Wrap(ctx, err, "adding fk reference in software to software_titles")
 }
 
+func (ds *Datastore) upsertSoftwareInstallerLabels(ctx context.Context, installerID uint, labels fleet.LabelIndentsWithScope) error {
+	var exclude bool
+	switch labels.LabelScope {
+	case fleet.LabelScopeIncludeAny:
+		exclude = false
+	case fleet.LabelScopeExcludeAny:
+		exclude = true
+	default:
+		return errors.New("invalid label scope")
```

Comment:
This should error only if there are labels to upsert (once we call this for all cases, even when no labels are provided).